tensorflow/contrib/gdr/gdr_memory_manager.h

```diff
@@ -0,0 +1,38 @@
+#ifndef GDR_MEMORY_MANAGER_H_
+#define GDR_MEMORY_MANAGER_H_
+
+#include "tensorflow/core/lib/core/status.h"
+
+namespace google {
+namespace protobuf {
+class Any;
+}
+}
+
+namespace tensorflow {
+
+class Device;
+class DeviceContext;
+class Tensor;
+
+class RemoteMemoryManager {
+ public:
+  virtual Status Init() = 0;
+  virtual void Run() = 0;
+  virtual void Stop() = 0;
+
+  virtual Status TransportOptionsFromTensor(
+      ::google::protobuf::Any* mutable_transport_options, const Tensor& tensor,
+      Device* device, DeviceContext* device_context, bool on_host) = 0;
+
+  virtual Status TensorFromTransportOptions(
+      Tensor* tensor, const ::google::protobuf::Any& transport_options,
+      Device* device, DeviceContext* device_context, bool on_host) = 0;
+};
+
+RemoteMemoryManager* CreateRemoteMemoryManager(const string& host,
```

Comment:
It will fail in the `GdrMemoryManager::Init()`, which will break in [`rdma_create_ep`](https://github.com/red-bird/tensorflow/blob/955764316dd9c796db874ae2034e336a48b1fb46/tensorflow/contrib/gdr/gdr_memory_manager.cc#L204) with `EADDRNOTAVAIL` (Cannot assign requested address). 

As a side note, the rdmacm library supports set `RDMA_OPTION_ID_REUSEADDR` using `rdma_set_option`, similar to `SO_REUSEADDR`. I could not think of an use case setting that option though, nor have I tried it myself.